.github/workflows/vm-kernel.yaml

```diff
@@ -87,25 +150,42 @@ jobs:
       - name: get kernel version
         id: get-kernel-version
         run: |
-          linux_config=$(ls neonvm/hack/linux-config-*)  # returns something like "neonvm/hack/linux-config-6.1.63"
-          kernel_version=${linux_config##*-}             # returns something like "6.1.63"
+          linux_config=$(ls neonvm/hack/kernel/linux-config-*)  # returns something like "neonvm/hack/kernel/linux-config-6.1.63"
+          kernel_version=${linux_config##*-}                    # returns something like "6.1.63"
 
           echo VM_KERNEL_VERSION=$kernel_version >> $GITHUB_OUTPUT
 
+      - name: get docker tags
+        id: get-tags
+        env:
+          KERNEL_VERSION_TAG: ${{ inputs.tag || steps.get-kernel-version.outputs.VM_KERNEL_VERSION }}
+          CACHE_TAG: ${{ needs.check-kernel-image.outputs.last-kernel-sha }}
+        run: |
+          # A comma-separated list of tags
+          TAGS="${VM_KERNEL_IMAGE}:${KERNEL_VERSION_TAG}"
+          TAGS="${VM_KERNEL_IMAGE}:${CACHE_TAG},${TAGS}"
+          TAGS="${VM_KERNEL_IMAGE}:${GITHUB_RUN_ID},${TAGS}"
+
+          echo "tags=${TAGS}" >> $GITHUB_OUTPUT
+
+          # `docker/build-push-action@v3` returns all ${TAGS} in metadata ("image.name" field), so it can't be used a image name right away.
+          # Choose one of them as a "canonical" tag and use it to construct the job output (along with a digest provided by `docker/build-push-action@v3`).
+          echo "canonical=${VM_KERNEL_IMAGE}:${GITHUB_RUN_ID}" >> $GITHUB_OUTPUT
+
       - name: build linux kernel
         id: build-linux-kernel
         uses: docker/build-push-action@v3
         with:
           build-args: KERNEL_VERSION=${{ steps.get-kernel-version.outputs.VM_KERNEL_VERSION }}
-          context: neonvm/hack
+          context: neonvm/hack/kernel
           platforms: linux/amd64
           # Push kernel image only for scheduled builds or if workflow_dispatch/workflow_call input is true
           push: ${{ github.event_name == 'schedule' && 'true' || ( inputs.push || 'false' ) }}
```

Comment:
If we're using this for caching, we now want to push always, right? (and I guess we can get rid of the schedule)